reconcile/cna/assets/aws_rds.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+from typing import Optional
+
+from pydantic import Field
+from pydantic.dataclasses import dataclass
+
+from reconcile.cna.assets.asset import (
+    Asset,
+    AssetError,
+    AssetModelConfig,
+    AssetStatus,
+    AssetType,
+)
+from reconcile.cna.assets.aws_utils import aws_role_arn_for_module
+from reconcile.gql_definitions.cna.queries.cna_resources import (
+    CNARDSInstanceDefaultsV1,
+    CNARDSInstanceV1,
+)
+
+
+@dataclass(frozen=True, config=AssetModelConfig)
+class AWSRDSAsset(Asset[CNARDSInstanceV1, CNARDSInstanceDefaultsV1]):
+    identifier: str = Field(alias="identifier")
+    vpc_id: str = Field(alias="vpc_id")
+    role_arn: str = Field(alias="role_arn")
+    db_subnet_group_name: str = Field(alias="db_subnet_group_name")
+    instance_class: str = Field(alias="instance_class")
+    allocated_storage: str = Field(alias="allocated_storage")
+    max_allocated_storage: str = Field(alias="max_allocated_storage")
+    engine: Optional[str] = Field(None, alias="engine")
+    engine_version: Optional[str] = Field(None, alias="engine_version")
+    major_engine_version: Optional[str] = Field(None, alias="major_engine_version")
+    username: str = Field(None, alias="username")
+    region: Optional[str] = Field(None, alias="region")
+    backup_retention_period: Optional[int] = Field(
+        None, alias="backup_retention_period"
+    )
+    backup_window: Optional[str] = Field(None, alias="backup_window")
+    maintenance_window: Optional[str] = Field(None, alias="maintenance_window")
+    multi_az: Optional[bool] = Field(None, alias="multi_az")
+    deletion_protection: Optional[bool] = Field(None, alias="deletion_protection")
+    apply_immediately: Optional[bool] = Field(None, alias="apply_immediately")
+
+    # Those values are implicit and not set in app-interface
```

Comment:
i agree that `is_production` is a good thing for direct users of CNA. when we talk about app-interface, where we also have defaults and overrides, it can get messy to know where the actual settings for an RDS instance are coming from. 

i would tend to ignore the `is_production` settings, providing an arbitrary value for it (in this case `true`), while at the same time making all parameters in the defaults schema mandatory so looking at app-interface is self-sufficient to get the full picture about configuration settings. we can provide meaningful presets as referegcable defaults datafiles.

wdyt?

Reply:
I think that the usage of `defaults` in app-interface isn't great today. Teams tend to copy them without much thought. The values are only as good as the person who created the file. The defaults provided by `is_production` are better in my (slightly biased) opinion because they are opinionated, and allow for the ability to quickly determine where teams have deviated from these default values.

I look at it like:

* `is_production` provides your sane defaults for most use cases
* `defaults` allows teams to avoid redundant configurations across services

The issue of where values come from is going to be present whether we hide `is_production` or not.

The module documentation might help clarify things a bit as well.